test/fuzz/fuzzerutil_test.cpp

```diff
@@ -0,0 +1,1519 @@
+               OpName %18 "f2"
+               OpName %20 "cf1"
+               OpName %22 "cf2"
+               OpName %26 "i1"
+               OpName %28 "i2"
+               OpName %30 "ci1"
+               OpName %32 "ci2"
+               OpName %36 "value"
+               OpDecorate %26 RelaxedPrecision
+               OpDecorate %28 RelaxedPrecision
+               OpDecorate %30 RelaxedPrecision
+               OpDecorate %32 RelaxedPrecision
+               OpDecorate %36 Location 0
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeBool
+          %7 = OpTypePointer Function %6
+          %9 = OpConstantTrue %6
+         %11 = OpConstantFalse %6
+         %14 = OpTypeFloat 32
+         %15 = OpTypePointer Function %14
+         %17 = OpConstant %14 1.23000002
+         %19 = OpConstant %14 1.11000001
+         %21 = OpConstant %14 2
+         %23 = OpConstant %14 3.29999995
+         %24 = OpTypeInt 32 1
+         %25 = OpTypePointer Function %24
+         %27 = OpConstant %24 1
+         %29 = OpConstant %24 100
+         %31 = OpConstant %24 123
+         %33 = OpConstant %24 1111
+         %35 = OpTypePointer Input %14
+         %36 = OpVariable %35 Input
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+          %8 = OpVariable %7 Function
+         %10 = OpVariable %7 Function
+         %12 = OpVariable %7 Function
+         %13 = OpVariable %7 Function
+         %16 = OpVariable %15 Function
+         %18 = OpVariable %15 Function
+         %20 = OpVariable %15 Function
+         %22 = OpVariable %15 Function
+         %26 = OpVariable %25 Function
+         %28 = OpVariable %25 Function
+         %30 = OpVariable %25 Function
+         %32 = OpVariable %25 Function
+               OpStore %8 %9
+               OpStore %10 %11
+               OpStore %12 %9
+               OpStore %13 %11
+               OpStore %16 %17
+               OpStore %18 %19
+               OpStore %20 %21
+               OpStore %22 %23
+               OpStore %26 %27
+               OpStore %28 %29
+               OpStore %30 %31
+               OpStore %32 %33
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_4;
+  const auto consumer = nullptr;
+  const std::unique_ptr<opt::IRContext> context =
+      BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  opt::IRContext* ir_context = context.get();
+  // A 32 bit float constant with width 14 exists and the id is 17.
+  // ASSERT_EQ(17, fuzzerutil::MaybeGetFloatConstant(ir_context,
+  // transformation_context, std::vector<uint32_t> {123}, 32, false));

```

Comment:
Actually those are the questions that I have regarding what word vector would be for floats and negative constants. They were the remaining questions I had in the previous PR. 
I assume word vector would entail information about the constant's value. Thus words vector for positive single constant would be that number exactly, but then I'm not sure how it would represent float and negative. Floats have different number of decimal places and negative constants need a number to indicate that its negative.

Reply:
Ah, makes sense!

For floats and negative values, the words should contain the bit pattern for the value.

You can achieve this via memcpy.

Check out line 875 of transformation_replace_constant_with_uniform_test.cpp, which does this:

```
  uint32_t float_data[1];
  float temp = 3.0;
  memcpy(&float_data[0], &temp, sizeof(float));
```

This causes float_data[0] to end up containing the bit pattern for the floating point number 3.0 (which is 0x40400000 according to [this converter](https://gregstoll.com/~gregstoll/floattohex/)).

Similarly, for a negative number you can memcpy from int32_t to uint32_t.

Note that in the above it would also be fine to do:

```
  uint32_t float_data;
  float temp = 3.0;
  memcpy(&float_data, &temp, sizeof(float));
```

The reason I used a 1 element array is for uniformity, as in other tests, for 64-bit floats, I use a 2-element array.

Reply:
There is also a function for this: `fuzzerutil::FloatToWord`.

Reply:
Cool - @sliu-UIUC please use that instead.

Reply:
Resolved.